samplers/aws/xray/remote_sampler.go

```diff
@@ -0,0 +1,240 @@
+	// List of known centralized sampling rules
+	manifest *centralizedManifest
+
+	// proxy is used for getting quotas and sampling rules
+	xrayClient *xrayClient
+
+	// pollerStart, if true represents rule and target pollers are started
+	pollerStart bool
+
+	// samplingRules polling interval, default is 300 seconds
+	samplingRulesPollingInterval time.Duration
+
+	// Unique ID used by XRay service to identify this client
+	clientID string
+
+	// Provides system time
+	clock Clock
+
+	mu sync.RWMutex
+}
+
+// Compile time assertion that remoteSampler implements the Sampler interface.
+var _ sdktrace.Sampler = (*RemoteSampler)(nil)
+
+// NewRemoteSampler returns a centralizedSampler which decides to sample a given request or not.
+func NewRemoteSampler(ctx context.Context, opts ...SamplerOption) (*RemoteSampler, error) {
+	options := new(samplerOptions).applyOptionsAndDefaults(opts...)
+
+	// set global logging
+	globalLogger = options.logger
+
+	// Generate clientID
+	var r [12]byte
+
+	_, err := crypto.Read(r[:])
+	if err != nil {
+		return nil, fmt.Errorf("unable to generate client ID: %w", err)
+	}
+
+	id := fmt.Sprintf("%02x", r)
+
+	clock := &DefaultClock{}
+
+	m := &centralizedManifest{
+		rules: []*centralizedRule{},
+		index: map[string]*centralizedRule{},
+		clock: clock,
+	}
+
+	remoteSampler := &RemoteSampler{
+		pollerStart:                  false,
+		clock:                        clock,
+		manifest:                     m,
+		clientID:                     id,
+		xrayClient:                   newClient(options.proxyEndpoint),
+		samplingRulesPollingInterval: options.samplingRulesPollingInterval,
+	}
+
+	remoteSampler.mu.Lock()
+	if !remoteSampler.pollerStart {
+		remoteSampler.start(ctx)
+	}
+	remoteSampler.mu.Unlock()
+
+	return remoteSampler, nil
+}
+
+func (rs *RemoteSampler) ShouldSample(parameters sdktrace.SamplingParameters) sdktrace.SamplingResult {
+	// ToDo: add business logic for remote sampling
+
+	return sdktrace.SamplingResult{}
+}
+
+func (rs *RemoteSampler) Description() string {
+	return "remote sampling with AWS X-Ray"
+}
+
+func (rs *RemoteSampler) start(ctx context.Context) {
+	if !rs.pollerStart {
+		rs.startRulePoller(ctx)
+	}
+
+	rs.pollerStart = true
+}
+
+func (rs *RemoteSampler) startRulePoller(ctx context.Context) {
+	go func() {
+		// Period = 300s, Jitter = 5s
+		t := newTicker(rs.samplingRulesPollingInterval, 5*time.Second)
+
+		// Initial refresh
+		if err := rs.refreshManifest(ctx); err != nil {
+			globalLogger.Printf("Error occurred while refreshing sampling rules. %v\n", err)
+		} else {
+			globalLogger.Println("Successfully fetched sampling rules")
+		}
+
+		// Periodic manifest refresh
+		for {
+			if err := rs.refreshManifest(ctx); err != nil {
+				globalLogger.Printf("Error occurred while refreshing sampling rules. %v\n", err)
+			} else {
+				globalLogger.Println("Successfully fetched sampling rules")
+			}
+			select {
+			case _, more := <-t.C():
+				if !more {
+					break
```

Comment:
```suggestion
					return
```

This is clearer and will survive more potential changes.